src/debug_user_code.py

```diff
@@ -56,12 +62,26 @@
 utils.abs_path_to_user_file = abs_path_to_code_file
 utils.debug_mode = True
 
+# overriding print function so that it shows on clue terminal
+def print_decorator(func):
+    global curr_terminal
+
+    def wrapped_func(*args, **kwargs):
+        curr_terminal.add_str_to_terminal("".join(str(e) for e in args))
+        return func(*args, **kwargs)
+
+    return wrapped_func
+
+
+print = print_decorator(print)
```

Comment:
do we need to restore print in the end?

Reply:
That's us using the print_decorator function to override print with our custom version of print.

Reply:
should we restore it though after debug_user_code is ran?
so like
original_print = print
print = print_decorator(print)

...
print = original_print

Reply:
In what case would this needed? When the program is done, all globals are reset.

Reply:
I guess so. Thanks Andrea